### DISCUSSION TITLE
Why do I get this error? application=indexer fetcher=block_catchup first_block_number=6 last_block_number=0 [error] ** (Postgrex.Error) ERROR 23505 (unique_violation) duplicate key value violates unique constraint "one_consensus_child_per_parent" and Key (parent_hash)=(\x0000000000000000000000000000000000000000000000000000000000000000) already exists ?

### DISCUSSION THREAD

**Discussion by Anilkumar18** [blockscout/blockscout#8713] [Q&A]
Key (parent_hash)=(\x0000000000000000000000000000000000000000000000000000000000000000) already exists.
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:803: Ecto.Adapters.SQL.insert_all/9
    (ecto 3.10.2) lib/ecto/repo/schema.ex:59: Ecto.Repo.Schema.do_insert_all/7
    (explorer 5.2.0) lib/explorer/repo.ex:61: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
    (elixir 1.14.5) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (explorer 5.2.0) lib/explorer/chain/import.ex:303: Explorer.Chain.Import.insert_changes_list/3
    (stdlib 4.0.1) timer.erl:235: :timer.tc/1
    (explorer 5.2.0) lib/explorer/prometheus/instrumenter.ex:17: Explorer.Prometheus.Instrumenter.block_import_stage_runner/4

2023-10-18T15:17:08.395 application=indexer fetcher=block_catchup first_block_number=6 last_block_number=0 [error] ** (Postgrex.Error) ERROR 23505 (unique_violation) duplicate key value violates unique constraint "one_consensus_child_per_parent"

    table: blocks
    constraint: one_consensus_child_per_parent

Key (parent_hash)=(\x0000000000000000000000000000000000000000000000000000000000000000) already exists.
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:803: Ecto.Adapters.SQL.insert_all/9
    (ecto 3.10.2) lib/ecto/repo/schema.ex:59: Ecto.Repo.Schema.do_insert_all/7
    (explorer 5.2.0) lib/explorer/repo.ex:61: anonymous fn/5 in Explorer.Repo.safe_insert_all/3
    (elixir 1.14.5) lib/enum.ex:2468: Enum."-reduce/3-lists^foldl/2-0-"/3
    (explorer 5.2.0) lib/explorer/chain/import.ex:303: Explorer.Chain.Import.insert_changes_list/3
    (stdlib 4.0.1) timer.erl:235: :timer.tc/1
    (explorer 5.2.0) lib/explorer/prometheus/instrumenter.ex:17: Explorer.Prometheus.Instrumenter.block_import_stage_runner/4

**Comment by sl1depengwyn** [comment]
The reason is that there should be no blocks with the same `parentHash` field on the chain, however your chain has such blocks
<img width="533" alt="image" src="https://github.com/blockscout/blockscout/assets/53992153/42b0788f-c820-4e1b-af82-b09789b1a673">
Example of how it looks on ETH Mainnet
<img width="554" alt="image" src="https://github.com/blockscout/blockscout/assets/53992153/b98ddb9d-fe8f-4ddf-a86f-75303dcdfda3">
